app/utils.ts

```diff
@@ -178,10 +178,10 @@ export function getCSSVar(varName: string) {
  * Detects if the Operation system is MacOS
  */
 export function isMacOS(): boolean {
-    if (window !== 'undefined') {
-        let userAgent = window?.navigator?.userAgent
-        if (userAgent.indexOf('Mac') != -1) return true
-    }
+  if (typeof window !== "undefined") {
+    let userAgent = window?.navigator?.userAgent;
+    if (userAgent.indexOf("Mac") != -1) return true;
```

Comment:
must refactor this instead of `Mac` its should be `macintosh` with parse lowerstring

Reply:
recommended ref code snippet : 
```ts
export function isMacOS(): boolean {
  const userAgent = navigator.userAgent.toLowerCase();
  const isMac = /(iphone|ipad|ipod|macintosh)/i.test(userAgent);
  const isTouchDevice = userAgent.includes("mac") && "ontouchend" in document;
  
  return isMac && isTouchDevice;
}
```

Reply:
@H0llyW00dzZ  Yooo. thank you so much for reviewing it! it helps me improve, that's very kind of you & I really love it. Honestly, you're a legend!


I'm pretty sure, we don't need to detect touchend events, right?

Reply:
@lifeeric  other mac like macbook pro with touch bar

Reply:
@H0llyW00dzZ If we already found out that it is a macintosh. we still need `ontouched` event since we don't use it?

Thank you

Reply:
@lifeeric Just in case, for the benefit of others, not just us.

Reply:
@H0llyW00dzZ please have a look again. thanks!